nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Discover/SdkProjectDiscovery.cs

```diff
@@ -203,6 +208,56 @@ public static async Task<ImmutableArray<ProjectDiscoveryResult>> DiscoverWithBin
                                 }
                             }
                             break;
+                        case Target target when target.Name == "_HandlePackageFileConflicts":
+                            // this only works if we've installed the exact SDK required
+                            if (experimentsManager.InstallDotnetSdks)
+                            {
+                                var projectEvaluation = GetNearestProjectEvaluation(target);
+                                if (projectEvaluation is not null)
+                                {
+                                    var removedReferences = target.Children.OfType<RemoveItem>().FirstOrDefault(r => r.Name == "Reference");
+                                    var addedReferences = target.Children.OfType<AddItem>().FirstOrDefault(r => r.Name == "Reference");
+                                    if (removedReferences is not null && addedReferences is not null)
+                                    {
+                                        foreach (var removedAssembly in removedReferences.Children.OfType<Item>())
+                                        {
+                                            var removedPackageName = GetChildMetadataValue(removedAssembly, "NuGetPackageId");
+                                            var removedFileName = Path.GetFileName(removedAssembly.Name);
+                                            if (removedPackageName is not null && removedFileName is not null)
```

Comment:
Can we maybe flip a couple of these conditions and have them `continue;` to reduce nesting?